### ISSUE TITLE
[Bug]. Iceberg DataSet from Flink NameSpace Different than Spark

### ISSUE THREAD

**Issue by FranMorilloAWS** [OpenLineage/OpenLineage#2625]
IcebergSinkVisitor registers Iceberg table with a different naming convention than Spark, in which it creates different lineage graph. Potential solution is adding Symlink in IcebergSink Visitor to same naming convention as spark

**Comment by dolfinus** [comment]
Could you give examples of dataset names procuded by both integrations?

**Comment by FranMorilloAWS** [comment]
This is how Flink leaves it as default
Namespace: s3://iceberg-open-lineage-<>
Name: /iceberg-2/open_lineage.db/test_symlink
Description: No Description
Spark leaves it as
Namespace: s3://iceberg-open-lineage-<>
Name: /iceberg-2/open_lineage.db/open_lineage.test_symlink

![image](https://github.com/OpenLineage/OpenLineage/assets/85679820/54e66d30-75bc-4f41-9592-4604ca2c09b4)

**Comment by FranMorilloAWS** [comment]
Both location are the S3 bucket. What is different is the namespace.  I believe is that IcebergSink Visitor is not creating the facet, so that it can also be seen as the same dataset for spark iceberg table

**Comment by pawel-big-lebowski** [comment]
> What is different is the **namespace**.

Your previous comments suggest the namespace is the same `Namespace: s3://iceberg-open-lineage-<>` for Flink and Spark. Is the problem related to dataset namespace or dataset name? 

If the problem is related to S3 path, then only one of them is valid. Could you point whether it is Spark or Flink integration pointing to incorrect location?

**Comment by FranMorilloAWS** [comment]
The problem is related to dataset name.  Even though there are both interacting with the same Iceberg Table. As the Iceberg Sink doesnt have a Facet. It draws the lineage as two different datasets, instead of being able to tell they are the same

**Comment by pawel-big-lebowski** [comment]
Got it. Dataset name is `/iceberg-2/open_lineage.db/test_symlink` or `/iceberg-2/open_lineage.db/open_lineage.test_symlink` are pointing to physical locations. However, only one of them is valid. Which one of them is valid (containing data)? Answering this will help figuring out if the issue is Spark or Flink related.

**Comment by FranMorilloAWS** [comment]
So Flink  has the right path.  Spark is adding the database name in front of the table name

 
![iceberg-openlieange](https://github.com/OpenLineage/OpenLineage/assets/85679820/a2f817c6-5e6f-418e-ae38-d58b50c29a28)

**Comment by FranMorilloAWS** [comment]
What we are seeing is that the Table Identifier that is given by Iceberg API is different that the one that Spark can access through the MetaStore.

**Comment by pawel-big-lebowski** [comment]
This can be the issue within this line: https://github.com/OpenLineage/OpenLineage/blob/1.16.0/integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/catalog/IcebergHandler.java#L90

**Comment by FranMorilloAWS** [comment]
Sowe believe that the namespace given in Spark doesnt include the folder of the iceberg database (openlineage.db) but it modifies it so its no the physical location of the iceberg table but it just then adds the database as.  db.tablename

An option is adding a facet to IcebergSink Visitor, but again we havent found a way to properly get the same datasetidentifier in flink as its done in spark. We could do a string replace ("db/","") but that feels like a hack.

**Comment by pawel-big-lebowski** [comment]
Wouldn't `identifer.name()` instead of `identifier.toString` be enough?

**Comment by FranMorilloAWS** [comment]
Yes but flink is storing the datasetidentifier using Table.location instead of building the uri itself,  also the URI path coming from the table location it still has the ".db" folder

**Comment by FranMorilloAWS** [comment]
Any news on this??

**Comment by FranMorilloAWS** [comment]
but iceberg visitor in flink would need to be changed so it registers it like that

**Comment by dolfinus** [comment]
> but iceberg visitor in flink would need to be changed so it registers it like that

Don't get it, both Spark and Flink are working with `org.apache.iceberg.Table` class which has `.location()` method.

**Comment by FranMorilloAWS** [comment]
flink registers the table in the dataset using a different convention than spark. So marquez when it builds the lineage its not able to identify that they are both the same dataset

**Comment by dolfinus** [comment]
Do Spark & Flink both store data in the same s3 location, but just report different format?

**Comment by FranMorilloAWS** [comment]
yes they both store in same s3 location, but when registering the dataset for openlineage they phrase differently

This is how Flink leaves it as default
Namespace: s3://<bucket-name>
Name: /iceberg-2/open_lineage.db/test_symlink
Description: No Description
Spark leaves it as
Namespace: s3://<bucket-name>
Name: /iceberg-2/open_lineage.db/open_lineage.test_symlink

**Comment by FranMorilloAWS** [comment]
Spark is adding to the db name to the table name in marquez

**Comment by FranMorilloAWS** [comment]
any news?

**Comment by dolfinus** [comment]
I can take is after finishing #2767

**Comment by dolfinus** [comment]
I've tried to create few Iceberg tables with local warehouse, and location is different from your screenshot.

```python
from pyspark.sql import SparkSession

spark = (
    SparkSession.builder
    .appName("PySpark SQL basic example")
    .config("spark.jars.packages", "org.apache.iceberg:iceberg-spark-runtime-3.4_2.12:1.5.0")
    .config("spark.some.config.option", "some-value")
    .config("spark.sql.catalog.ice_local", "org.apache.iceberg.spark.SparkCatalog")
    .config("spark.sql.catalog.ice_local.type", "hadoop")
    .config("spark.sql.catalog.ice_local.warehouse", "file:/tmp/iceberg")
    .getOrCreate()
)

spark.sql("USE ice_local")
spark.sql("CREATE DATABASE mydb")
spark.sql("CREATE DATABASE parent.child")
spark.sql("CREATE TABLE nodb (id INT)")
spark.sql("CREATE TABLE mydb.withdb (id INT)")
spark.sql("CREATE TABLE parent.child.complexdb (id INT)")
```
```bash
$ ls -lsah /tmp/iceberg/**
/tmp/iceberg/nodb
/tmp/iceberg/mydb/withdb/metadata
/tmp/iceberg/parent/child/complexdb
```

There is no `.db` suffix in database folder name.

**Comment by dolfinus** [comment]
I'm talking about this screenshot:
https://github.com/OpenLineage/OpenLineage/issues/2625#issuecomment-2133184192

**Comment by FranMorilloAWS** [comment]
Not sure why the diference. Im wondering if using the Glue Data Catalog and S3 would be the culprit.  Also the main issue is at the table name in which even though the paths in S3 are the same. Spark openlineage adds the database to the table name as well

**Comment by dolfinus** [comment]
Could you provide Spark session config and example of events produces by OpenLineage integration?

**Comment by dolfinus** [comment]
I've implemented #2797. But if real table's location is different in Spark and Flink, this will lead to producing datasets with different names
